src/Accessor/DefaultAccessorStrategy.php

```diff
@@ -72,14 +72,18 @@ public function getValue(object $object, PropertyMetadata $metadata, Serializati
             return $object->{$metadata->getter}();
         }
 
-        if ($metadata->forceReflectionAccess) {
+        if ($metadata->forceReflectionAccess || ($context->getShouldSerializeUnitializedAsNull() && $metadata->hasType)) {
             $ref = $this->propertyReflectionCache[$metadata->class][$metadata->name] ?? null;
             if (null === $ref) {
                 $ref = new \ReflectionProperty($metadata->class, $metadata->name);
                 $ref->setAccessible(true);
                 $this->propertyReflectionCache[$metadata->class][$metadata->name] = $ref;
             }
 
+            if (PHP_VERSION_ID >= 70400 && !$ref->isInitialized($object)) {
```

Comment:
Should it be executed also for the cases when `$metadata->forceReflectionAccess` is set to true, but `$context->getShouldSerializeUnitializedAsNull()` will return false?